newIDE/app/src/ObjectEditor/Editors/SpriteEditor/CollisionMasksEditor/CollisionMasksPreview.js

```diff
@@ -92,11 +189,57 @@ const CollisionMasksPreview = (props: Props) => {
     return { frameX, frameY };
   };
 
+  const addVertex = (newVertexHintPoint: NewVertexHintPoint) => {
+    const vertices = props.polygons
+      .at(newVertexHintPoint.polygonIndex)
+      .getVertices();
+    const verticesSize = vertices.size();
+    const newVertex = new gd.Vector2f();
+    newVertex.x = newVertexHintPoint.x;
+    newVertex.y = newVertexHintPoint.y;
+    vertices.push_back(newVertex);
+    newVertex.delete();
+    vertices.moveVector2fInVector(verticesSize, newVertexHintPoint.vertexIndex);
+    const vertex = vertices.at(newVertexHintPoint.vertexIndex);
+    setDraggedVertex({
+      vertex,
+      polygonIndex: newVertexHintPoint.polygonIndex,
+      vertexIndex: newVertexHintPoint.vertexIndex,
+    });
+    setNewVertexHintPoint(null);
+    props.onClickVertice(vertex.ptr);
+    props.onPolygonsUpdated();
+  };
+
+  /**
+   * @returns true if the vertex should be deleted.
+   */
+  const magnetDraggedVertexForDeletion = (): boolean => {
```

Comment:
I think this function should have a clearer name to say what it does.
But it's confusing because it does both "update the draggedVertex" and "say if it should be deleted"
maybe call it `updateMagnetDraggedVertexForDeletion()`

Reply:
I though is was a verb, I'll replace if with "magnetize".